### PR TITLE
React to Nudge, and fix marquee

### PR DESCRIPTION
**Pull request by volfied**
f9a3db3 addresses #2150, and it works. However, PS doesn't always send us ALL the nudge events, so if you want to test it, please download latest Jenkins of pg-dev to get ALL the nudge events.

40c6003 addresses #2301, problem was, I was only finishing marquee select if there was a rectangle being drawn, so for small enough distances / fast mouse events / action not finishing on time, the marquee flag would remain up, and overlays would be broken. Now it's more stable, and we reset the flag on tool select as well (so user can't switch tools while marquee selecting to leave the flag up, darn user!)

### CODE REVIEW

File: src/js/actions/superselect.js [hunk 1]
```diff
@@ -703,6 +703,10 @@ define(function (require, exports) {
      */
     var marqueeSelect = function (doc, ids, add) {
         this.dispatch(events.ui.SUPERSELECT_MARQUEE, { enabled: false });
+
+        if (!ids) {
```
Comment (iwehrman) [inline src/js/actions/superselect.js]: If `ids` can be null, the JSDoc param needs to be updated with `?Array...`

**Comment by iwehrman** [comment]
Just the JSDoc nit.

**Comment by iwehrman** [comment]
Oh, wait, also: don't you want to remove the `nudgeLayers` action?

**Comment by volfied** [comment]
Thanks @iwehrman, made changes.

**Comment by iwehrman** [comment]
:+1:

**Pull request merged**